File: Easy/Rock Paper Scissors Lizar Spock/Solutions with Input from txt File/SolutionOOP.java
import java.io.FileNotFoundException;
import java.io.File;
import java.util.*;

class SolutionOOP {

    // Which symbols win against which - first winner and second loser
    private static List <String> pattern = Arrays.asList("CP", "CL", "PR", "PS", "RL", "RC", "LS", "LP", "SC", "SR");

    private static class Player {
        // States
        private int number;
        private String sign; 
        private ArrayList<Integer> enemies = new ArrayList <Integer>(); 
    
        // Constructor
        private Player(int num, String sign){
            this.number = num;
            this.sign = sign;
        }

        // Behaviours
        public void addEnemy(int idEnemy){ 
            this.enemies.add(idEnemy);
        }

        public String enemyString(){
            String s = "";
            for(Integer i : enemies){
                s += String.valueOf(i) + " ";
            }

            return s.substring(0, s.length() - 1);
        }
    }

    public static void main(String args []) throws
FileNotFoundException {

        File input = new File("Test Sets/test1.txt"); // Paste Source path of the txt file of the input
        Scanner in = new Scanner(input);

        int n = in.nextInt(); // The number of players
        Queue <Player> competition = new LinkedList <> (); 

        for(int i = 0; i < n; i++){
            int np = in.nextInt();
            String sp = in.next();
            competition.add(new Player(np, sp));
        }
        in.close();

        while (competition.size() > 1){ 
            // Duel the first 2 player in the competition and add the winner to the end of the queue
            competition.add(Duel(competition.poll(), competition.poll()));
        }

        Player winner = competition.poll(); 
        System.out.println();
        System.out.println("The winner was Player: " + winner.number); 
        System.out.println("This Player went up against the following players: " + winner.enemyString()); 
        System.out.println();
    }

    private static Player Duel(Player p1, Player p2){
        p1.addEnemy(p2.number);
        p2.addEnemy(p1.number);

        if(pattern.contains(p1.sign + p2.sign)){ 
            return p1;
        } else if(pattern.contains(p2.sign + p1.sign)){ 
            return p2; 
        } else if(p1.number < p2.number){ 
            return p1; 
        } else 
            return p2; 
    }
}

File: Easy/Rock Paper Scissors Lizar Spock/Solutions with Input from txt File/SolutionOOP.py
# Which symbols win against which - first winner and second loser
pattern = ["CP", "CL", "PR", "PS", "RL", "RC", "LS", "LP", "SC", "SR"]

with open("Test Sets/test1.txt")  as file: # Paste Source path of the txt file of the input
    lines = file.readlines()
line = [x[:-1] for x in lines[:-1]]
line.append(lines[-1])

class Player:
    # States
    enemies = ""

    # Constructor
    def __init__ (self, num, sign):
        self.number = num
        self.sign = sign
    
    # Behaviours
    def addEnemy(self, enemy): 
        self.enemies += str(enemy.number) + " "

    def enemyString (self): 
        return self.enemies[0:-1]

def Duel (p1, p2):
    Player.addEnemy(p1, p2)
    Player.addEnemy(p2, p1)

    if (p1.sign + p2.sign in pattern): 
        return p1
    elif (p2.sign + p1.sign in pattern): 
        return p2 
    elif (p1.number < p2.number): 
        return p1 
    else: 
        return p2 

n = int(line[0]) # The number of players
competition = [] 

for i in range(n):
    inputs = line[1+i].split()
    np = int(inputs[0])
    sp = inputs[1]
    competition.append(Player(np, sp))

while len(competition) > 1: 
    # Duel the first 2 player in the competition and add the winner to the end of the queue
    competition.append(Duel(competition.pop(0), competition.pop(0))) 

winner = competition.pop(0)
print()
print("The winner was Player: ", winner.number)
print("This Player went up against the following players: ", Player.enemyString(winner)) 
print()